libs/dh/shared/feature-flags/src/lib/feature-flags.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { DhAppEnvironments } from '@energinet-datahub/dh/shared/environments';
+
+export type DhFeatureFlag = {
+  created: string;
+  disabledEnvironments: DhAppEnvironments[];
```

Comment:
suggestion: the essential purpose for using feature flags, to me, is to "enable the intentional development, deployment and release of new functionality without affecting the existing one". When I look that, I think having an "enabled unless disabled" approach seems too loose. When it comes to "hiding" something behind a feature flag, it should be as hard as possible to accidentally "leak" it unintentionally. On the other hand, the act of "revealing" it  should be as intentional as possible.

The problem with `disabledEnvironments` is that forgetting to add an env to the list accidentally releases "the thing". Adding a new environment (or chaining the name of an existing one) will lead to the same result. 

Having said that, I think our approach should be the opposite, i,e, `enabledInEnvironments`. It will be more verbose but less error prone. Forgetting to add an env to the list will do nothing, "the thing" will not be released.

What do you think?

Reply:
I see your point, however as you said it would properly be more verbose since I guess it typically should be enabled for dev and test but disabled for prod. Also the fact that "the users" should actively use the feature-flags functionality, and if they don't it will also be released (maybe by accident) and that would be pretty hard for us to guard us against that?

Reply:
> Also the fact that "the users" should actively use the feature-flags functionality, and if they don't it will also be released (maybe by accident) 

How will something be released by accident in this case?

The way I see it, having a `disabled unless enabled` approach means that "the user" must intentionally add an env to the list in order to enable "the thing" for that env. So releasing something by accident in that case means adding an env to the list that shouldn't be there (either intentionally or not). In the opposite approach (i.e. "enabled unless disabled") , NOT adding an env will mean releasing "the thing". 

So is "adding something [extra] to a list" vs "forgetting to add something to a list" more error prone? I'd argue it's the latter.

As for verbosity, it seems like it's the same. In one case is adding "dev" and "test" env, in the other case is adding "pre-prod" and "prod" env.

Reply:
We keep it as is, and gather feedback from the users.